### PR TITLE
powered addition solution

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

the editorial is kinda crap ngl, would appreciate any feedback on that

### CODE REVIEW

**Review by envyaims — APPROVED** [review]
you implemented it so weird but ok

**Comment by SansPapyrus683** [comment]
how's it weird lmao

File: solutions/silver/cf-1338A.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+If we have a number that's less than its predecessor, there's *no reason*
+to make that number greater than what comes before it.
+
+With that out of the way, we know how much we should add to each number.
+To get the overall minimum number of seconds to make the sequence nondecreasing,
+we can take the floors of the log base 2 of each number and add 1.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n)$ for each test case.
```
Comment (envyaims) [inline solutions/silver/cf-1338A.mdx:23]: log function takes log n time no?

**Comment by envyaims** [comment]
> how's it weird lmao

theres a much straight forward way to implement it?

```c++
void JoeMama(){
	int n; cin>>n;
	vector<int> nums(n);
	for(int& i: nums)cin>>i;
	int ans = 0;
	for(int i = 1; i < n; i++){
		if(nums[i] < nums[i-1]){
			int diff = nums[i-1] - nums[i];
			int lg = log2(diff);
			ans = max(ans,lg);
			nums[i] = nums[i-1];
		}
	}
	cout<<ans + 1<<"\n";
}
```

**Pull request merged**